api/capi/src/nnstreamer-capi-single.c

```diff
@@ -119,6 +118,78 @@ ml_single_get_tensors_info_from_filter (GstElement * filter, gboolean is_input,
   g_free (prop_prefix);
 }
 
+/**
+ * @brief Internal static function to get the string of tensor-filter properties.
+ * The returned string should be freed using g_free().
+ */
+static gchar *
+ml_single_get_filter_option_string (gboolean is_input,
+    ml_tensors_info_s * tensors_info)
+{
+  GstTensorsInfo gst_info;
+  gchar *str_prefix, *str_dim, *str_type, *str_name;
+  gchar *option;
+
+  ml_tensors_info_copy_from_ml (&gst_info, tensors_info);
+
+  if (is_input)
+    str_prefix = g_strdup ("input");
+  else
+    str_prefix = g_strdup ("output");
+
+  str_dim = gst_tensors_info_get_dimensions_string (&gst_info);
+  str_type = gst_tensors_info_get_types_string (&gst_info);
+  str_name = gst_tensors_info_get_names_string (&gst_info);
+
+  option = g_strdup_printf ("%s=%s %stype=%s %sname=%s",
+      str_prefix, str_dim, str_prefix, str_type, str_prefix, str_name);
+
+  g_free (str_prefix);
+  g_free (str_dim);
+  g_free (str_type);
+  g_free (str_name);
+  gst_tensors_info_free (&gst_info);
+
+  return option;
+}
+
+/**
+ * @brief Internal static function to set tensors info in handle.
```

Comment:
Internal static function to set tensors info in **the** handle

Reply:
updated